Automattic/pocket-casts-ios
# Search Improvements: add results view

Fixes #
| 📘 Project: #709 |
|:---:|

Add the search results view.

Please note that the podcast carousel is not exactly like what is contained in the Figma designs. I had a hard time reproducing that and decided to move on and tackle that later.

https://user-images.githubusercontent.com/7040243/220983844-ecf3d757-035c-459a-8499-ad6de2af888d.mp4

## To test

1. Run the app
2. Go to Profile > Settings > Beta Features > enable `newSearch`
3. Go to Discover
4. Tap the search input
5. ✅ Search results should appear
6. Type any search term on the search input
7. ✅ The search results should appear after a while
8. Tap "X" on search input
9. ✅ Search history should appear
10. Tap "Cancel"
11. ✅ The search is dismissed

## Checklist
